tower-sessions-core/src/session.rs

```diff
@@ -376,10 +376,16 @@ impl Session {
     /// assert_eq!(session.get::<usize>("foo").await.unwrap(), Some(42));
     /// # });
     /// ```
+    #[tracing::instrument(skip(self))]
     pub async fn clear(&self) {
         let mut record = self.record.lock().await;
         if let Some(record) = record.as_mut() {
+            tracing::trace!("clearing record");
             record.data.clear();
+        } else {
+            tracing::trace!("record not loaded from store; creating a new one to clear");
+            *self.session_id.lock() = None;
```

Comment:
We don't want to reset the session ID. Instead this should establish a new record with the current ID. In the case of no session ID, there's nothing we need to do with the record directly.

It would also be nice to update the doc tests to cover the case where we've called clear on a session which contains data but hasn't loaded via get_record yet.

Reply:
Sure thing, I agree keeping the id is better but since create_record() generates a random one I didn't think of just resetting it after creating it. Your change works for me.